### PR TITLE
feat: Introduce IntoCStr trait

### LINKED ISSUE #18: CStr in public APIs
libnv's current API expects `&str` for all names and string-like values.  It then internally converts them to `CString`.  But sometimes it's necessary to insert a name or value that's already a `&CStr`, for example when a C function calls into Rust.  It would be more efficient and ergonomic if libnv could accept those, too.  Something like this:
```rust
pub trait IntoCStr {
    fn into_cstr(self) -> NvResult<Cow<CStr>> {
        ...
    }
}
impl IntoCStr for &str {...}
impl IntoCStr for &CStr {...}
impl NvList {
    fn insert_string<N: IntoCStr, V: IntoCStr>(&mut self, name: N, value: V) -> NvResult<()> {
        ...
    }
}
```

### PR DESCRIPTION
**Pull request by andoriyu**
Make public interface more ergonomic by accepting more than just `&str`
in public interface.

BREAKING CHANGE: Public interface now uses generic type that implements
`IntoCStr` instead of `&str` as its input. Function signatures has
changed, but code should work as is.

BREAKING CHANGE: `libnv::insert_strings` now takes ownership of the
value and no longer panics if conversion failed. From now on, it will
return first error as a result.

Closes https://github.com/Inner-Heaven/libnv-rs/issues/18

### CODE REVIEW

**Comment by andoriyu** [comment]
@asomers Is this what you had in mind?

File: src/nvpair.rs [hunk 1]
```diff
@@ -891,4 +935,21 @@ mod test {
...
     }
+
+    #[test]
+    fn into_c_str() {
+        let owned = CString::new("owned").unwrap();
+        let mut list = NvList::new(NvFlag::UniqueNameType).unwrap();
+        list.insert(owned, 1u32).unwrap();
+
+        let borrowed_name = CString::new("borrowed").unwrap();
+        let borrowed = unsafe { CStr::from_ptr(borrowed_name.as_ptr()) };
```
Comment (asomers) [inline src/nvpair.rs]:
```suggestion
        let borrowed: &CStr = borrowed_name.as_c_str();
```

Comment (asomers) [reply src/nvpair.rs]: No need for unsafe here.

**Review by asomers — COMMENTED** [review]
Yeah, this is what I had in mind.

**Pull request merged**